### DISCUSSION TITLE
I can't get my Gotek to recognize my flash drive

### DISCUSSION THREAD

**Discussion by JeffH2022** [keirf/flashfloppy#727] [Q&A]
OK, here is my setup.  I have a TRS80 - model 3, FD0 is a real floppy drive that seems to work fine, FD1 is the Gotek.  I flashed the flashfloppy firmware and it says F-F when I boot.  I took a flashdrive and used the USB_Foppy_Manager_v1.4i to format the flash and
create 100 virt disks.  I copied FF.CFG in to the root (it put it in DISK00) and put a DSK file in DISK00 as well.  When I boot the TRS80
the Gotek get power and settles to 000.  However  if I do a dir :0 I see the bootdisk (real disk) directory, but if I do a dir :1 I
get an illegal drive error.  On a prior flash I also got a E01 error.  What am I doing wrong?  It seems like the problem is in the 
flashdrive (by the way, I did setup the FF.CFG explicitly to shugart)?

Thanks.

**Comment by keirf** [comment]
Don't use the management software that's for factory firmware. Instead flashfloppy works with image files on a fat formatted usb drive. So you need to copy some compatible image files onto a USB drive.

**Comment by JeffH2022** [comment]
I tried just copying it into a formatted flashdrive and it's still giving me an 'illegal drive number error'  when I try to do a dir :1The green LED lights up and the LED display flashes a few numbers and then settles to '000'BTW, if you don't use virt disks how do you distinguish between image files?
Thanks.

    On Sunday, December 11, 2022 at 11:31:31 AM CST, Keir Fraser ***@***.***> wrote:  
 
 


Don't use the management software that's for factory firmware. Instead flashfloppy works with image files on a fat formatted usb drive. So you need to copy some compatible image files onto a USB drive.

**Reply by keirf** [reply]
Number your files 000.img 001.img etc (or whatever file extension is correct for trs80 image files) and since they are sorted into alphanumeric order they will automatically be assigned to slots 000 001 and so on.

**Reply by keirf** [reply]
Beyond that your problem is likely that your image files are not compatible for this system. Or if they are maybe try converting to hfe image format using hxc tools. You will need to Google up a bit on trs80 gotek setup.

**Comment by JeffH2022** [comment]
Yep, I tried that.  I have 4 files on my flash drive..2 dsk images, 1 ff.cfg, and an img.cfgI still get an 'illegal drive number' error on the model 3 if I do a 'dir :1'The odd thing is I see the Gotek being accessed prior to the error (i.e. the green led turns on, andthe red leds change numbers, etc).  I have no idea why/what's not working.
    On Sunday, December 11, 2022 at 11:31:31 AM CST, Keir Fraser ***@***.***> wrote:  
 
 


Don't use the management software that's for factory firmware. Instead flashfloppy works with image files on a fat formatted usb drive. So you need to copy some compatible image files onto a USB drive.

**Comment by keirf** [comment]
There are plenty of reasons it can not work, since if any part of setup isn't correct then of course it doesn't work at all. Perhaps the host isn't correctly set up? https://torlus.com/floppy/forum/viewtopic.php?t=1871

**Comment by JeffH2022** [comment]
Hi Keir,  Sorry if this isn't the right email but can I ask a couple questions about an "older" version of Grease Weazel?
Thanks,-Jeff
    On Sunday, December 11, 2022 at 12:25:53 PM CST, Keir Fraser ***@***.***> wrote:  
 
 


Number your files 000.img 001.img etc (or whatever file extension is correct for trs80 image files) and since they are sorted into alphanumeric order they will automatically be assigned to slots 000 001 and so on.

**Comment by keirf** [comment]
Yes

**Comment by JeffH2022** [comment]
Actually I was going to ask via email but you answered on FB

Yahoo Mail: Search, Organize, Conquer 
 
  On Thu, Apr 11, 2024 at 1:35 AM, Keir ***@***.***> wrote:   

Yes

**Comment by plaes** [comment]
@keirf Could you also add mention to wiki (https://github.com/keirf/flashfloppy/wiki/Initial-Setup#usb-storage) that drive needs to be partitioned using MBR partitions. (At least disks formatted with gnome-disks under Fedora 40 seem to use GPT by default).

**Reply by keirf** [reply]
FlashFloppy can read GPT partitioned drives.

**Reply by plaes** [reply]
Weird, I had also issues getting drive recognized by FF, so today I got a  known working USB drive from a friend, saw that it was using MBR partitions and after converting my flash drive to use MBR things started working.

**Reply by keirf** [reply]
Yeah that's weird, it definitely should work with GPT. Though AUTOBOOT.HFE (hosted selector program for Amiga, Atari ST) doesn't.

**Reply by keirf** [reply]
Is it at all possible that your partition wasn't formatted FAT32?

**Comment by JeffH2022** [comment]
Cool.  But what does that help with exactly?  

Yahoo Mail: Search, Organize, Conquer 
 
  On Thu, Oct 10, 2024 at 2:07 PM, Keir ***@***.***> wrote:   


FlashFloppy can read GPT partitioned drives.

**Reply by keirf** [reply]
Reading a drive that has a GPT rather than a MBR.

**Comment by JeffH2022** [comment]
Yeah but isn't the main advantage to GPT bigger drives?  If I'm using flashfloppy on a gotek and have a limit of 1000 floppy images isn't that 1000 x 1.5M ~ 1.5G which is doable with MBR?  Maybe I'm wrong...which is why I'm asking

Yahoo Mail: Search, Organize, Conquer 
 
  On Thu, Oct 10, 2024 at 3:26 PM, Keir ***@***.***> wrote:   


Reading a drive that has a GPT rather than a MBR.

**Reply by keirf** [reply]
Oh I see. It's just the modern default partition scheme on many operating systems, and it's easy enough to support in the firmware. But there is no particular benefit to using GPT on FlashFloppy you are right.

**Comment by JeffH2022** [comment]
I'm also debating to trying on it on something other than gotek, for instance I have another emulator that uses SD cards...haven't figured that out yet and just got gotek working reliably.  What I'd love is an emulator that would look like a real-time drive (i.e. not just work with static image files, etc).  But I'm still a FTE so my hobby time is limited.

Yahoo Mail: Search, Organize, Conquer 
 
  On Thu, Oct 10, 2024 at 3:56 PM, Keir ***@***.***> wrote:   


Is it at all possible that your partition wasn't formatted FAT32?
